tools/utils.py

```diff
@@ -112,6 +112,16 @@ def delete_contents(dirname, exclude=None):
       delete_file(entry)
 
 
+def get_num_cores():
+  # Prefer `process_cpu_count` when available (3.13 and above). since
+  # it takes into account thread affinity.
+  if hasattr(os, 'process_cpu_count'):
+    cpu_count = os.process_cpu_count()
```

Comment:
You could also add an implementation that uses `len(os.sched_getaffinity())` where available (that also doesn't work everywhere but should cover some OSes on older Pythons.).